hack/generator/pkg/jsonast/jsonast.go

```diff
@@ -420,7 +420,7 @@ func refHandler(ctx context.Context, scanner *SchemaScanner, schema *gojsonschem
 
 		description := "Generated from: " + url.String()
 
-		sd := astmodel.NewStructDefinition(structReference, structType.Fields()...).WithDescription(&description)
+		sd := astmodel.NewStructDefinition(structReference, structType).WithDescription(&description)
```

Comment:
Agreed, wrapping the entire Struct Type makes more sense than extracting the fields.